docs/source/model/dependencies.rst

```diff
@@ -177,8 +177,87 @@ The manually defined dependencies will override the default ones MLflow detects
     test your model in a virtual environment. Please refer to :ref:`Validating Environment for Prediction <validating-environment-for-prediction>` for more details.
 
 
-Saving Extra Code with an MLflow Model
---------------------------------------
+Saving Extra Code dependencies with an MLflow Model - Automatic inference
+-------------------------------------------------------------------------
+
+.. note::
+    Automatic code dependency inference is a feature that was introduced in MLflow 2.13.0 and is marked as Experimental. The base implementation may be 
+    modified, improved, and adjusted with no prior notice in order to address potential issues and edge cases. 
+
+.. note::
+    Automatic code dependency inference is currently supported for Python Function Models only. Support for additional named model flavors will be coming in 
+    future releases of MLflow.
+
+.. warning::
+    Before using dependency inference, ensure that your dependent code modules do not have sensitive data hard-coded within the modules (e.g., passwords, 
+    access tokens, or secrets). Code inference does not obfuscate sensitive information and will capture and log (save) the module, regardless of what it contains. 
+
+In the MLflow 2.13.0 release, a new method of including custom dependent code was introduced that expands on the existing feature of declaring ``code_paths`` when 
+saving or logging a model. This new feature utilizes import dependency analysis to automatically infer the code dependencies required by the model by checking which 
+modules are imported within the references of a Python Model's definition. 
+
+In order to use this new feature, you can simply set the argument ``infer_code_paths`` (Default ``False``) to ``True`` when logging. You do not have to define 
```

Comment:
Note: currently the PR only supports MLflow python model (by mlflow.pyfunc.log_model / save_model), I will file follow-up PR for  supporting all other flavors